### PR TITLE
(Fix) Update DApp loading code considering upcoming Breaking Change in MetaMask (Branch Master)

### LINKED ISSUE #1165: (Fix) Update DApp loading code considering upcoming Breaking Change in MetaMask
**Problem:** On `November 2nd, 2018` MetaMask and other DApp browsers will stop exposing user accounts by default. Instead, dapps must request access using a new provider method: `provider.enable()`. Detailed info is available here: https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8

**Solution:** Fix the code which loads DApp according to instructions from `What this means for developers` section of the article mentioned above.

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #1165 .

Modify the "MetaMask not found" error page to display a generic error text stating "Wallet not found, or access to Ethereum account not granted"

This is for the master branch, there will be another PR for the branch 2.0

screenshot_20181012_152956

### CODE REVIEW

**Comment by vbaranov** [comment]
@mariano-aguero

Double dots:
screen shot 2018-10-16 at 13 54 55

and let's change the link to Token Wizard wiki

**Comment by mariano-aguero** [comment]
@vbaranov 

> Double dots:
and let's change the link to Token Wizard wiki

Done!

**Comment by mariano-aguero** [comment]
@dennis00010011b 

> Manage page isn't fully rendered unless clicking button 'Finalize'

It is not an error related to the changes made it, anyway I solved in this branch

Regards

File: src/react-web3/Web3Provider.js [hunk 1]
```diff
@@ -9,30 +10,76 @@ class Web3Provider extends Component {
...
+    }.bind(this), 1500)
+  }
+
+  checkWeb3() {
+    const setWeb3 = () => {
+      window.web3 = new Web3(window.web3.currentProvider)
+      this.setState({
+        web3: window.web3
+      })
+      this.fetchAccounts()
```
Comment (fernandomg) [inline src/react-web3/Web3Provider.js]: Here `fetchAccounts` is being called, but it's called aswell inside the `interval` defined in the line `91`, which is called after calling `checkWeb3`, in the line `40`

File: src/react-web3/Web3Provider.js [hunk 2]
```diff
@@ -12,72 +12,95 @@ class Web3Provider extends Component {
...
-          // Request account access if needed
-          ethereum.enable().then(()=> {
-            if (!this.state.web3) {
-              setWeb3()
-            }
-          })
-        } catch (error) {
-          // User denied account access...
-          console.log('User denied account access')
+    return new Promise((resolve, reject) => {
```
Comment (fernandomg) [inline src/react-web3/Web3Provider.js]: Why a `Promise` with `throw` and `reject`/`resolve` mixed?

Can it be done entirely with `async/await` or `resolve/reject`?

File: src/react-web3/Web3Provider.js [hunk 3]
```diff
@@ -12,72 +12,95 @@ class Web3Provider extends Component {
...
+        // Non-dapp browsers...
+        else {
+          const msj = 'Non-Ethereum browser detected. You should consider trying a wallet!'
+          throw new Error(msj)
+        }
+      } catch (err) {
+        this.setState({
+          approvePermissions: false
+        })
+        console.log(err.message)
```
Comment (fernandomg) [inline src/react-web3/Web3Provider.js]: `logger`

Comment (mariano-aguero) [reply src/react-web3/Web3Provider.js]: @fernandomg 
If my memory doesn't fail me, `logger` is used in the branch 2.0, please correct me if I'm wrong

Greetings

**Comment by dennis00010011b** [comment]
I can't get "MetaMask not found" error page with local build. As a result app crashed on Step 2.
https://www.useloom.com/share/8d421012f09940f291ad75c79fb91015

screen shot 2018-10-19 at 10 35 48 am

**Comment by mariano-aguero** [comment]
@dennis00010011b 

> I can't get "MetaMask not found" error page with local build. As a result app crashed on Step 2.

Fixed

**Pull request merged**